EIPS/eip-2375.md

```diff
@@ -0,0 +1,118 @@
+---
+eip: 2375
+title: ENS support for canonical names
+author: Ilan Olkies (@ilanolkies)
+discussions-to: https://discuss.ens.domains/t/new-standard-proposal-ens-cname-support/1236
+status: Draft
+type: Standards Track
+category: ERC
+created: 2019-11-11
+---
+
+## Simple Summary
+ENS CNAME-like resolver.
+
+## Abstract
+This EIP specifies a resolver for canonical name resolution for ENS. This permits identifying the same resource with several names.
+
+## Motivation
+In existing systems, resources often have several names that identify the same resource.  For example, the names *mew.eth* and *myetherwallet.eth* may both identify the same address. Similarly, in the case of mailboxes, many organizations provide many names that actually go to the
+same mailbox.
+
+Most of these systems have a notion that one of the equivalent set of names is the canonical or primary name and all others are aliases.
+
+DNS provides such a feature using the canonical name (CNAME) RR. A CNAME RR identifies its owner name as an alias, and specifies the corresponding canonical name in the RDATA section of the RR.
+
+## Specification
+
+A resolver supporting `cname` interface identifies its name as an alias, and specifies the corresponding canonical name. It must provide a fallback function that:
+
+1. Fetches the canonical name of the queried name.
+4. Replaces the canonical name in the call data.
+2. Fetches the canonical name's resolver.
+3. Queries the canonical name's resolver.<sup>1</sup>
+
+If a `cname` resolution is present at a node, no other resolution should be present; this ensures that the data for a canonical name and its aliases cannot be different.
+
+Domain names which point at another name should always point at the primary name and not the alias. This avoids extra indirections in
+accessing information.
+
+By the robustness principle, domain resolution should not fail when presented with `cname` chains or loops; `cname` chains should be followed and `cname` loops signalled as an error.
```

Comment:
Maybe I'm missing something, but this doesn't look nearly specific enough to be a standard.  It seems to just describe an idea in high level terms.  It describes a fallback function that does some stuff and doesn't return anything, which suggests that the "stuff" it does is just an implementation detail and not relevant to the spec, at which point this spec is pretty meaningless.

Reply:
The initial idea was specifying a _cname_ standard record that was queried after a 0 response from a resolver. Then, it evolved into a resolver implementing this “substitution”. The discussion was held in ENS forum and was lost.